File: backend/app/schemas/stripe.py
from typing import Optional

from pydantic import BaseModel


class CheckoutRequest(BaseModel):
    plan: str  # "solo" or "team"
    billing: str  # "monthly" or "yearly"


class CheckoutResponse(BaseModel):
    url: str


class PortalResponse(BaseModel):
    url: str


class CancelRequest(BaseModel):
    immediate: bool = False


class SubscriptionStatusResponse(BaseModel):
    plan: str
    status: str
    period_end: Optional[str] = None
    created_at: Optional[str] = None
    is_refund_eligible: bool = False


class CancelResponse(BaseModel):
    status: str
    period_end: Optional[str] = None
    message: str


class RefundResponse(BaseModel):
    status: str
    amount: Optional[int] = None
    currency: Optional[str] = None
    message: str

File: frontend/src/lib/api/stripe.ts
import { api } from './axios-instance'

export const stripeApi = {
  createCheckoutSession: (plan: string, billing: string) =>
    api.post<{ url: string }>('/api/stripe/create-checkout-session', { plan, billing }),

  createPortalSession: () => api.post<{ url: string }>('/api/stripe/customer-portal'),

  getSubscriptionStatus: () =>
    api.get<{
      plan: string
      status: string
      period_end: string | null
      created_at: string | null
      is_refund_eligible: boolean
    }>('/api/subscription/status'),

  cancelSubscription: (immediate: boolean) =>
    api.post<{ status: string; period_end: string | null; message: string }>(
      '/api/subscription/cancel',
      { immediate },
    ),

  requestRefund: () =>
    api.post<{ status: string; amount: number | null; currency: string | null; message: string }>(
      '/api/subscription/refund',
    ),
}
